--- ../phpurs/src/Phpurs/Printer.purs
+++ ../phpurs/src/Phpurs/Printer.purs
@@ -100,7 +100,11 @@
       argNames = joinWith ", " (map (\a -> "$" <> sanitize a) args)
       nStr = show (length args)
+      nArgs = length args
       
-      useClause = if length captures > 0 then " use (" <> joinWith ", " captures <> ", &$__fn)" else " use (&$__fn)"
+      useClause = if nArgs == 1 then
+                    (if length captures > 0 then " use (" <> joinWith ", " captures <> ")" else "")
+                  else
+                    (if length captures > 0 then " use (" <> joinWith ", " captures <> ", &$__fn)" else " use (&$__fn)")
       
       rewrittenStmts = replaceReturn stmts
@@ -145,15 +149,20 @@
 
       fnBody = 
         "  $__num = func_num_args();\n" <>
-        "  if ($__num < " <> nStr <> ") {\n" <>
-        fastPathStr <>
-        "    return phpurs_curry_fallback($__fn, func_get_args(), " <> nStr <> ");\n" <>
-        "  }\n" <>
+        (if nArgs == 1 then "" else
+        "  if ($__num < " <> nStr <> ") {\n" <>
+        fastPathStr <>
+        "    return phpurs_curry_fallback($__fn, func_get_args(), " <> nStr <> ");\n" <>
+        "  }\n") <>
         (if length rewrittenStmts > 0 then "  " <> joinWith ";\n  " (map printExpr rewrittenStmts) <> ";\n" else "") <>
         "  __end:\n" <>
         "  return $__num > " <> nStr <> " ? $__res(...array_slice(func_get_args(), " <> nStr <> ")) : $__res;\n"
 
     in 
-      "(function()" <> useClause <> " {\n" <>
-      "  $__fn = function(" <> argStr <> ")" <> useClause <> " {\n" <> fnBody <> "  };\n" <>
-      "  return $__fn;\n" <>
-      "})()"
+      if nArgs == 1 then
+        "function(" <> argStr <> ")" <> useClause <> " {\n" <> fnBody <> "}"
+      else
+        "(function()" <> useClause <> " {\n" <>
+        "  $__fn = function(" <> argStr <> ")" <> useClause <> " {\n" <> fnBody <> "  };\n" <>
+        "  return $__fn;\n" <>
+        "})()"
